monai/transforms/lazy/utils.py

```diff
@@ -38,7 +38,7 @@ def is_affine_shaped(data):
             return False
         if not hasattr(data, "shape") or len(data.shape) < 2:
             return False
-        return data.shape[-1] in (3, 4) and data.shape[-2] in (3, 4) and data.shape[-1] == data.shape[-2]
+        return data.shape[-1] in (3, 4) and data.shape[-1] == data.shape[-2]
```

Comment:
Hi @wyli  and @KumoLiu , nhere I simplified the logic. However, I'm not sure if the whole logic is correct.
For example, if the input is a tensor that has the shape `(100, 1, 3, 3)`, it meets the logic, but how can we ensure that it is an affine matrix?
Similar question on `DisplacementField `, it seems any tensor with `len(data.shape) >= 3`, and not meet `is_affine_shaped ` will be considered as a  dense displacement field.

Feel free to correct me if I misunderstand anything here, and if the input here has some other restrictions and can only belong to `Affine` or `DisplacementField`, let me add some descriptions to avoid misunderstanding. Thanks!

Reply:
Yes, the change is good for now, we can enhance it when we deal with https://github.com/Project-MONAI/MONAI/issues/4604

Reply:
I see, thanks @wyli , thus we can keep this version so far.